client/src/App.js

```diff
@@ -3,12 +3,42 @@ import ViewEntries from './ViewEntries';
 import './reset.css';
 import './layout.css';
 import './styles.css';
+import ViewCreateEnty from './ViewCreateEntry';
+import { useState } from 'react';
+/* exported data */
+
+let data = {
+  view: 'entry-form',
+  entries: [],
+  editing: null,
+  nextEntryId: 1,
+};
+
+// window.addEventListener('beforeunload', function (event) {
+//   const dataJSON = JSON.stringify(data);
+//   localStorage.setItem('code-journal-data', dataJSON);
+// });
+
+// const localData = JSON.parse(localStorage.getItem('code-journal-data'));
+// if (localData) {
+//   data = localData;
+// }
 
 function App() {
+  const [view, setView] = useState('entry-form');
+
+  function handleSwap(text) {
+    setView(text);
```

Comment:
You probably don't want to require the subcomponents to know what values represent the views ('entry-form' and 'entries'). Instead, just have them let you know that they need to be swapped, and you set `view` to the proper value.